src/solvers/smt2_incremental/smt_sorts.def

```diff
@@ -0,0 +1,2 @@
+SORT_ID(bool)
```

Comment:
I honestly don't see the value of using this over `ID_bool` used in `bool_typet`.

Reply:
Please bear in mind that this file is not only being used to define the IDs of the `smt_sortt` classes. It is also being used to automate writing some of the more repetitive parts of the visitor pattern. :thinking: I should probably add some doxygen to this file explaining how it is used. What would be the value in rationing the number of different IDs used rather than keeping them separate?